packages/to-json-schema/src/toJsonSchema.test.ts

```diff
@@ -46,6 +46,55 @@ describe('toJsonSchema', () => {
         },
       });
     });
+
+    test('for recursive schema', () => {
+      const ul = v.object({
+        type: v.literal('ul'),
+        children: v.array(v.lazy(() => li)),
```

Comment:
Changing this to `v.lazy(() => v.array(li))` still causes the error. I will investigate this now.

Reply:
This `v.lazy(() => v.array(li))` cannot work because the reference of `v.array(li)` is always newly created and therefore cannot be associated with any reference in our reference map. We need to document that the return type of the getter must be a static reference for recursive schemas to work.

Reply:
Oh yeah I forgot that this was the reason why I explicitly required schemas wrapped in a lazy to be present in the definitions in @gcornut/valibot-json-schema

Reply:
Another solution could be to memoize unwrapping lazy schemas 🤔